falcon/asgi/app.py

```diff
@@ -976,7 +976,13 @@ async def _handle_websocket(self, ver, scope, receive, send):
             #   we don't support, so bail out. This also fulfills the ASGI
             #   spec requirement to only process the request after
             #   receiving and verifying the first event.
-            await send({'type': EventType.WS_CLOSE, 'code': WSCloseCode.SERVER_ERROR})
+            await send(
+                {
+                    'type': EventType.WS_CLOSE,
+                    'code': WSCloseCode.SERVER_ERROR,
+                    'reason': 'Internal Server Error',
```

Comment:
here we should probably send reason only if the `ver` is 2.3 or greater.

This here poses a problem since doing a simple `ver >= "2.3"` does not work assuming future version, since that check would fail if ver ever goes over `"2.10"`.

we probably need to convert `ver` to a tuple of int, but I don't know if we can assume that it will always be in the form `xx.yy` and not have values like `2.3.1` or something. @vytas7 do you have suggestions here?

Reply:
Maybe we could split `ver` by `.` and check the numbers one by one?

Reply:
I added a new member function in asgi/ws.py to check if the asgi version support `reason`. Is it more reasonable to put it in util/misc.py so both asgi/ws.py and asgi/app.py can utilize this function?

Reply:
I think we could keep it in ws.py, but we could make `_check_support_reason` a function, so that app could call it too?

Reply:
Personally, I can accept `>= '2.3'` as good enough until proven otherwise; the ASGI WebSocket spec is not evolving _that_ fast and I'm not sure if we live to see 2.10 :smiling_imp: Moreover, they might cut straight to `3.x` which would still work with this simple comparison.

Reply:
since that's not too performance critical (happens once per web socket), I think being correct here is worth it since it's not that much more code

Reply:
The function `check_support_reason()` has been added in `ws.py`. Hopefully, it could deal with any form of `ver` string, whether it is `2.3` or `2.10.1`.